docs/customization/api.rst

```diff
@@ -0,0 +1,80 @@
+Customizing API
+===============
+
+For create all endpoints in Sylius API we are using the API Platform.
+API Platform allows to configure endpoint by ``yaml`` and ``xml`` files or in ``annotations``.
+In these guide we will show you how to customize Sylius API endpoint with ``xml``.
+
+How to prepare project to customization?
+========================================
+
+For any kind of customization firstly you need to configure location of your configs.
+
+.. code-block:: yaml
+    # config/packages/api_platform.yaml
+    api_platform:
+    mapping:
+        paths:
+            - '%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources'
+            - '%kernel.project_dir%/config/api_platform'
+    patch_formats:
+        json: ['application/merge-patch+json']
+    swagger:
+        versions: [3]
+
+And now if you want to customize any API resource you need copy entire configuration of this resource from ``%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/`` to ``%kernel.project_dir%/config/api_platform``
+
+How to add an additional endpoint to a resource?
+================================================
+
+Let's assume that you want to add new endpoint to ``order`` which will dispatch a command.
+You need to add proper configuration in ``config/api_platform/Order.xml``:
+
+.. code-block:: xml
+
+    <collectionOperations>
+        <collectionOperation name="custom_operation">
+            <attribute name="method">POST</attribute>
+            <attribute name="path">/shop/order/custom-operation</attribute>
+            <attribute name="messenger">input</attribute>
+            <attribute name="input">App\Command\CustomCommand</attribute>
+        </collectionOperation>
+    </collectionOperations>
+
+And that's all, now you have new endpoint with your custom logic.
+
+.. tip::
+
+    Read more about API Platform endpoint configuration `here <https://api-platform.com/docs/core/operations/>`_
+
+How to remove a resource endpoint?
+==================================
+
+Let's assume that your shop offer only digital products so you don't need setting shipment method at all.
+To remove endpoint you only need to delete unnecessary configuration from ``config/api_platform/Order.xml``
```

Comment:
Is this sentence true? I mean we have some underlying logic that changes states of cart and checkout. Speaking of endpoint only POV, well we can just remove the endpoint and go drink some margarita 🍸 , but I don't think we can finish checkout process 🙁

Reply:
yes, but actually no :D
if you have only one shipping method and enabled option `Skip shipping step if only one shipping method is available?` on the channel, while checkout you don't need to set shipping method :)
you can check how it works on UI :)